### PR TITLE
Feature 4 (WIP) 

### PR DESCRIPTION
**Pull request by deyviconde**
Feature 4 Work in progress:
- Update entry was added
- data JSON small bug fixed (with datetime)
- New entry was fixed

### CODE REVIEW

File: route.rb [hunk 1]
```diff
@@ -25,17 +25,25 @@
 # end
 
 post '/' do
+  puts params
```
Comment (sergiodxa) [inline route.rb]: Why do you decided to keep this `puts`?

File: controller/entry_methods.rb [hunk 2]
```diff
@@ -4,28 +4,32 @@
...
-  data = JSON.parse(data_file)
-  data
+ data_file = File.read($path_data)
+ data_file = "{}" if data_file == ""
+ data = JSON.parse(data_file)
+ data
 end
 
-def save_data(id, entry)
-  data[id] = entry
```
Comment (MariethPerez) [inline controller/entry_methods.rb]: Please explain me Why was this line eliminated?

File: controller/entry_methods.rb [hunk 3]
```diff
@@ -4,28 +4,32 @@
...
-  save_data(id, entry)
-end
+def update_data(id, title, content)
+ data_file = read_data
+ entry_edit = data_file[id]
+ entry_edit["content_before"] << [entry_edit["title"], entry_edit["content"]]
+ entry_edit["title"] = title
+ entry_edit["content"] = content
+ save_data(data_file)
+end
```
Comment (paulotijero) [inline controller/entry_methods.rb]: Thanks for refactoring code :100:

File: controller/entry_methods.rb [hunk 4]
```diff
@@ -4,28 +4,32 @@
...
 
 def read_data
-  data_file = File.read($path_data)
-  data_file = "{}" if data_file == "" 
-  data = JSON.parse(data_file)
-  data
+ data_file = File.read($path_data)
+ data_file = "{}" if data_file == ""
+ data = JSON.parse(data_file)
+ data
```
Comment (KattyaCuevas) [inline controller/entry_methods.rb]: The indentation is one space?

File: controller/entry_methods.rb (hunk 2 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: When we save data, we don't need id. We receive all data package ❌

File: public/css/style.css [hunk 5]
```diff
@@ -60,20 +62,6 @@ textarea {
...
-}
-
-.item {
-  width: 230px;
-  height: 200px;
-  object-fit: cover;
-  padding: 15px;
-  border: 1px solid black;
-}
-
```
Comment (paulotijero) [inline public/css/style.css]: Why did you decide to delete this block of code? :thinking:

File: controller/entry_methods.rb [hunk 6]
```diff
@@ -4,28 +4,32 @@
...
-def update_data(id, entry)
-  data_file = read_data
-  save_data(id, entry)
-end
+def update_data(id, title, content)
+ data_file = read_data
+ entry_edit = data_file[id]
+ entry_edit["content_before"] << [entry_edit["title"], entry_edit["content"]]
+ entry_edit["title"] = title
+ entry_edit["content"] = content
```
Comment (KattyaCuevas) [inline controller/entry_methods.rb]: How do you use `entry_edit`?

File: route.rb [hunk 7]
```diff
@@ -25,17 +25,25 @@
...
-  inputs["is_deleted"] = 0
-  inputs["deleted_datetime"] = ""
-  add_data(inputs)
-  @message = "La nueva entrada de titulo #{params[:title]} fue creado exitosamente"
+  unless params.has_key?("id")
+    inputs["datetime"] = params["date"] + " " + params["time"]
+    inputs["content_before"] = [] 
+    inputs["highlight"] = 0
+    inputs["is_deleted"] = 0
+    inputs["deleted_datetime"] = ""
```
Comment (KattyaCuevas) [inline route.rb]: This validation should be in other method.

File: route.rb [hunk 8]
```diff
@@ -25,17 +25,25 @@
...
   inputs["title"] = params["title"]
-  inputs["datetime"] = params["date"] + " " + params["time"]
   inputs["content"] = params["content"]
-  inputs["content_before"] = [] 
-  inputs["highlight"] = 0
-  inputs["is_deleted"] = 0
-  inputs["deleted_datetime"] = ""
-  add_data(inputs)
-  @message = "La nueva entrada de titulo #{params[:title]} fue creado exitosamente"
+  unless params.has_key?("id")
```
Comment (paulotijero) [inline route.rb]: I recommend moving all this code block to "list_methods.rb"

File: route.rb (hunk 1 above)
Comment (deyviconde) [reply route.rb]: Sorry, we were doing some tests 👀  and I forgot to delete it. I will fix it now, thanks !

File: views/entry.erb [hunk 9]
```diff
@@ -1,29 +1,40 @@
+require "date"
 <% content_for :entry do %>
-<form action="/" method="post">
+<form action="/<%="?id=" + @entry_json["id"] if @entry_json%>" method="post">
```
Comment (paulotijero) [inline views/entry.erb]: Nice!!

File: controller/entry_methods.rb (hunk 3 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: You're welcome dude !

File: views/entry.erb [hunk 10]
```diff
@@ -1,29 +1,40 @@
...
+          </label>
+          <%else%>
           <label
             ><input
               type="date"
               class="form-control"
               name="date"
-              style="width:150px; font-family: 'Questrial', serif;"/>
-              <input
+              style="width:150px; font-family: 'Questrial', serif;"
```
Comment (paulotijero) [inline views/entry.erb]: This could add it to a class "font-input", because it repeats

File: controller/entry_methods.rb (hunk 4 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: Yeah, you are right. I just noticed that I have my editor wrongly configured. Thanks Kattya !!

Comment (deyviconde) [reply controller/entry_methods.rb]: I will fix it now !!!

File: public/css/style.css (hunk 5 above)
Comment (deyviconde) [reply public/css/style.css]: We had an issue with one version, sorry. I will review it today

File: controller/entry_methods.rb (hunk 6 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: Instead of repeating always data_file[id] I prefer to use one word 💇‍♂️

File: route.rb (hunk 7 above)
Comment (deyviconde) [reply route.rb]: Yes, my teammate is working on it. We will move this block to a method a.s.a.p

File: route.rb (hunk 8 above)
Comment (deyviconde) [reply route.rb]: Yep, I will clean this block and move it to a function in order to reuse it

File: views/entry.erb (hunk 9 above)
Comment (deyviconde) [reply views/entry.erb]: Thanks 🇵🇪

File: controller/entry_methods.rb (hunk 6 above)
Comment (KattyaCuevas) [reply controller/entry_methods.rb]: 🤔 Is it working?

File: views/entry.erb (hunk 10 above)
Comment (deyviconde) [reply views/entry.erb]: Oh, I got your idea. Ok, cool, I will wrap it an move it to a CSS class. Thanks, dude

File: controller/entry_methods.rb (hunk 6 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: Yep, it's working 👀

File: route.rb [hunk 11]
```diff
@@ -25,28 +25,20 @@
...
-  inputs["title"] = params["title"]
-  inputs["datetime"] = params["date"] + " " + params["time"]
-  inputs["content"] = params["content"]
-  inputs["content_before"] = [] 
-  inputs["highlight"] = 0
-  inputs["is_deleted"] = 0
-  inputs["deleted_datetime"] = ""
-  add_data(inputs)
-  @message = "La nueva entrada de titulo #{params[:title]} fue creado exitosamente"
+  @message = validate_new_or_update_data(params)
```
Comment (deyviconde) [inline route.rb]: Love it !!!!

File: route.rb [hunk 12]
```diff
@@ -60,5 +52,17 @@
 end
 
 get "/edit" do
-  # TODO: Add edit logic
+  @entry_json = recover_element(params["id"])
```
Comment (deyviconde) [inline route.rb]: Thanks for refactoring it @Linzeur  !!!

File: controller/entry_methods.rb [hunk 13]
```diff
@@ -1,31 +1,31 @@
 require 'json'
-require "date"
```
Comment (liannivin) [inline controller/entry_methods.rb]: Why you deleted it?

File: controller/entry_methods.rb [hunk 14]
```diff
@@ -1,31 +1,31 @@
...
-
 $path_data = "./data/entries.json"
-
```
Comment (liannivin) [inline controller/entry_methods.rb]: Good.

File: route.rb (hunk 11 above)
Comment (liannivin) [inline route.rb]: Excellent. This change clears much the code.

File: views/entry.erb (hunk 9 above)
Comment (liannivin) [inline views/entry.erb]: Could you explain that, please?

File: controller/entry_methods.rb (hunk 13 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: We think it's unnecessary. We don't depend of "date"

File: controller/entry_methods.rb (hunk 14 above)
Comment (deyviconde) [reply controller/entry_methods.rb]: Thanks!!

File: route.rb (hunk 11 above)
Comment (deyviconde) [reply route.rb]: Yep, I know. Looks cleaner !!

File: views/entry.erb (hunk 9 above)
Comment (deyviconde) [reply views/entry.erb]: Yep, if @entry_json exists it @entry_json["id"] will enter to id

**Pull request merged**